lib/CustomFields/pages/EditCustomFieldsSettings/EditCustomFieldsSettings.js

```diff
@@ -72,23 +84,42 @@ const EditCustomFieldsSettings = ({
     });
   };
 
+  const updateTitle = data => {
+    const body = JSON.stringify({
+      module: backendModuleName.toUpperCase(),
+      configName: 'custom_fields_label',
+      value: data,
+    });
+
+    if (sectionTitle.id) {
+      return makeTitleRequest(`configurations/entries/${sectionTitle.id}`)({
+        method: 'PUT',
+        body,
+      });
+    }
+
+    return makeTitleRequest('configurations/entries')({
+      method: 'POST',
+      body,
+    });
+  };
+
   const fetchUsageStatistics = useCallback(fieldId => {
     return makeOkapiRequest(`custom-fields/${fieldId}/stats`)({
       method: 'GET',
     });
   }, [makeOkapiRequest]);
 
-  const saveCustomFields = async data => {
+  const saveCustomFields = async (data, accTitle) => {
     setSubmitting(true);
-    const response = await updateCustomFields(data);
-
-    if (response.ok) {
-      redirectToView();
-    } else {
-      showErrorNotification();
-      setSubmitting(false);
-      setSubmitFinished(true);
-    }
+
+    Promise.all([updateCustomFields(data), updateTitle(accTitle)])
+      .then(redirectToView)
+      .catch(() => {
+        showErrorNotification();
+        setSubmitting(false);
+        setSubmitFinished(true);
+      });
```

Comment:
Fetch rejects promises only when the request fails because of network-related issues.
If the response's status code is something like 404, `catch`'s callback won't be executed and the error won't be handled. 
You need to check if `response.ok` is `true` to make sure that the response was fulfilled successfully.